actix-service/src/boxed.rs

```diff
@@ -113,24 +110,24 @@ where
     }
 }
 
-struct ServiceWrapper<T: Service>(T);
+struct ServiceWrapper<S: Service<Req>, Req>(S, PhantomData<Req>);
 
-impl<T> ServiceWrapper<T>
+impl<S, Req> ServiceWrapper<S, Req>
 where
-    T: Service + 'static,
-    T::Future: 'static,
+    S: Service<Req> + 'static,
+    Req: 'static,
+    S::Future: 'static,
 {
-    fn boxed(service: T) -> BoxService<T::Request, T::Response, T::Error> {
-        Box::new(ServiceWrapper(service))
+    fn boxed(service: S) -> BoxService<Req, S::Response, S::Error> {
+        Box::new(ServiceWrapper(service, PhantomData))
     }
 }
 
-impl<T, Req, Res, Err> Service for ServiceWrapper<T>
+impl<S, Req, Res, Err> Service<Req> for ServiceWrapper<S, Req>
 where
-    T: Service<Request = Req, Response = Res, Error = Err>,
-    T::Future: 'static,
+    S: Service<Req, Response = Res, Error = Err>,
+    S::Future: 'static,
 {
-    type Request = Req;
     type Response = Res;
     type Error = Err;
     type Future = BoxFuture<Res, Err>;
```

Comment:
Could be using `type Future = LocalBoxFuture<Result<Self::Response, Self::Error>>;`